server/routes/classes.ts

```diff
@@ -58,35 +127,101 @@ router.put("/:id", async (req: Request, res: Response) => {
         return res.status(404).json({ error: 'Class not found' })
     }
 
+
     const { 
         title,
         description, 
         order,
-        file,
     }: ClassInputs = req.body
 
-    if (!title && !description && !order && !file) {
+    if (!title && !description && !order && !req.file) {
         return res.status(400).json({
             error: 'No fields to update provided'
         })
     }
 
+    // Option 1: Update file (and fields)
+    if (req.file) {
+        try{
+
+            if(title) _class.title = title
+            if(description) _class.description = description
+            if(order) _class.order = order
+
+                // Verify file type
+            
+            const contentType = req.file.mimetype;
+
+            if (!allowedMimeTypes.includes(contentType)) {
+                return res.status(400).json({
+                    error: "Invalid file type. Only video files are allowed.",
```

Comment:
Maybe it would be correct to put in the message what type of files are accepted (mp4, quicktime and mpeg)